compiler/src/iree/compiler/Codegen/Common/CombineLayoutTransformation.cpp

```diff
@@ -198,12 +210,159 @@ foldExtractSliceIntoMapScatter(RewriterBase &rewriter,
+/// from the direct consumer of the `mapScatterOp`, which is expected to be an
+/// `iree_codegen.store_to_memref` op. Return failure if the result buffer is
+/// not found.
+static FailureOr<MapScatterOp>
+foldPadIntoMapScatter(RewriterBase &rewriter, tensor::PadOp padOp,
+                      MapScatterOp mapScatterOp,
+                      DistributionConfigFn distributionConfigFn) {
+  // Find the output buffer that the mapScatterOp is stored into.
+  if (!mapScatterOp->hasOneUse()) {
+    return rewriter.notifyMatchFailure(
+        mapScatterOp, "map_scatter does not have a single user");
+  }
+  auto storeOp = dyn_cast<IREE::Codegen::StoreToMemrefOp>(
+      *mapScatterOp->getUsers().begin());
+  if (!storeOp) {
+    return rewriter.notifyMatchFailure(
+        mapScatterOp,
+        "map_scatter user is not an iree_codegen.store_to_memref op");
+  }
+  Value outputBuffer = storeOp.getTarget();
+
+  // Write the padding values directly into the outputBuffer.
+  rewriter.setInsertionPointAfter(storeOp);
+  Location loc = padOp->getLoc();
+  SmallVector<OpFoldResult> padSrcSizes =
+      tensor::getMixedSizes(rewriter, loc, padOp.getSource());
+  SmallVector<OpFoldResult> ubs =
+      tensor::getMixedSizes(rewriter, loc, padOp.getResult());
+  SmallVector<OpFoldResult> lbs(ubs.size(), rewriter.getIndexAttr(0));
+  SmallVector<DistributionConfig> distConfigs = distributionConfigFn(
+      padOp.getSourceType().getShape(), rewriter.getContext());
+
+  auto innerLoopBuilder = [&](OpBuilder &b, Location loopLoc, ValueRange ivs) {
+    // We need to scatter the padding values according to the existing
+    // mapScatterOp transformation, so clone the transformation into the
+    // loop nest.
+    auto clonedMapScatterOp = cast<MapScatterOp>(b.clone(*mapScatterOp));
+    Block &clonedTransformBody =
+        clonedMapScatterOp.getTransformationRegion().getBlocks().front();
+    // Get a pointer to the YieldOp before inlining the Block.
+    auto yieldOp =
+        cast<IREE::LinalgExt::YieldOp>(clonedTransformBody.getTerminator());
+    rewriter.inlineBlockBefore(&clonedTransformBody, clonedMapScatterOp, ivs);
+    rewriter.eraseOp(clonedMapScatterOp);
+    OpBuilder::InsertionGuard g(b);
+    b.setInsertionPointAfter(yieldOp);
+    // Compute the indices into the outputBuffer, and the if condition to
+    // write the padding values. Padding values must obey the existing mask
+    // of the current mapScatterOp, and also be in the low or high pad
+    // range of the padOp.
+    SmallVector<OpFoldResult> low = padOp.getMixedLowPad();
+    Value writeCond = nullptr;
+    for (auto [l, srcSize, idx] : llvm::zip_equal(low, padSrcSizes, ivs)) {
+      Value lowVal = getValueOrCreateConstantIndexOp(rewriter, loc, l);
+      auto isLowPad = rewriter
+                          .create<arith::CmpIOp>(loc, arith::CmpIPredicate::ult,
+                                                 idx, lowVal)
+                          ->getResult(0);
```

Comment:
nit: I think we don't need `->getResult(0)` and we can replace `auto` with `Value`. It looks simpler to me. (Same comment for the `isHighPad`.)

```suggestion
      Value isLowPad = rewriter.create<arith::CmpIOp>(
          loc, arith::CmpIPredicate::ult, idx, lowVal);
```